core/client/src/test/java/alluxio/client/file/policy/DeterministicHashPolicyTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.policy;
+
+import alluxio.Constants;
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.wire.WorkerNetAddress;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Tests {@link DeterministicHashPolicy}.
+ */
+public final class DeterministicHashPolicyTest {
```

Comment:
We can divide the functionality into a few tests:

`getWorkerConsistently`
`getWorkerWithEnoughSpace`
`getWorkerTwoShards`

each test should just focus on one aspect of `DeterministicHashPolicy` so that they are simple and easy to understand